togetherly/lib/views/widgets/chore_list.dart

```diff
@@ -0,0 +1,171 @@
+// ignore_for_file: constant_identifier_names
+
+import 'package:flutter/material.dart';
+import 'package:togetherly/models/chore.dart';
+import 'package:togetherly/themes.dart';
+
+enum RequestedChores {
+  TODAY,
+  UPCOMING,
+  OVERDUE
+}
+// toUpperCase() with title to match?
+
+class ChoreList extends StatefulWidget {
+  const ChoreList({super.key, required this.title});
+  final String title;
+
+  @override
+  State<ChoreList> createState() => _ChoreListState();
+}
+
+class _ChoreListState extends State<ChoreList> {
+  List<Chore> chores = [
+    Chore(
+      title: 'Do the dishes',
+      dueDate: DateTime(2024, 2, 15),
+      points: 15,
+      isBonus: false,
+    ),
+    Chore(
+      title: 'Pick up the living room',
+      dueDate: DateTime(2024, 2, 14),
+      points: 10,
+      isBonus: false,
+      status: ChoreStatus.PENDING,
+    ),
+    Chore(
+      title: 'Walk the dog',
+      dueDate: DateTime(2024, 2, 14),
+      points: 20,
+      isBonus: true,
+      status: ChoreStatus.COMPLETED,
+    ),
+  ];
+
+  String prettyDate(DateTime dueDate) {
+    return "${dueDate.month}/${dueDate.day}/${dueDate.year}";
+  }
+
+  DateTime getDateToday() {
+    DateTime date = DateTime.now();
+    return DateTime(date.year, date.month, date.day);
+  }
+
+  bool isDueToday(Chore chore) {
+    return chore.dueDate == getDateToday();
+  }
+
+  Widget getStatusIcon(Chore chore) {
+    switch (chore.status) {
+      case ChoreStatus.ASSIGNED:
+        return const Icon(
+          Icons.check_box_outline_blank,
+          size: 28,
+          color: AppColors.brandBlack,
+        );
+      case ChoreStatus.PENDING:
+        return const Icon(
+          Icons.hourglass_bottom,
+          size: 28,
+          color: AppColors.brandRose,
+        );
+      case ChoreStatus.COMPLETED:
+        return const Icon(
+          Icons.check_box_outlined,
+          size: 28,
+          color: AppColors.brandGreen,
+        );
+      default:
+        throw Exception('Invalid status for chore');
+    }
+  }
+
+  Widget buildChoreElements() {
+    return Column(
+      children: <Widget>[
+        for (final chore in chores)
+          Container(
```

Comment:
It might be good to pull out the individual chore element as its own widget (ChoreItem or something along those lines). Then you could drastically simplify this widget's build method, and inline this `buildChoreElements` method.

Reply:
You can do it as part of this PR. Just push new commits with the necessary changes.